### DISCUSSION TITLE
How to use with custom scrollbar library (e.g. simplebar, OverlayScrollbars, etc)?

### DISCUSSION THREAD

**Discussion by taviks** [TanStack/virtual#504] [Q&A] [answered]
In my app "virtual" was working perfectly... spent a good long while working out small issues. It's implemented using vanilla JS (well, Imba, actually). 

I'm wanting overlay scrollbars. Attempting to implement https://github.com/Grsmto/simplebar now but am having issues getting it to work with "virtual". Am also considering https://github.com/KingSora/OverlayScrollbars. 

Any insight on getting "virtual" to work with one of these?

**Comment by taviks** [comment]
Related: 

- https://github.com/KingSora/OverlayScrollbars/issues/499
- https://github.com/Grsmto/simplebar/issues/546

**Reply by piecyk** [reply]
@jaymakes11 one option could be https://codesandbox.io/s/pensive-bartik-x2ko08?file=/src/index.js that is real power of virtual as you are in control of markup.

**Comment by taviks** [comment, marked as answer]
This was much easier than I anticipated. 

The trick seems to be updating the `getScrollElement` callback from this:

```imba
getScrollElement: do() document.getElementById('my-scroll-container-id')
```

to this:

```imba
getScrollElement: do() simpleBar.getScrollElement()
```

**Comment by xmsz-stu** [comment]
when  i use SimpleBar, the scroll seems to be stuck

```
  const parentRef = useRef<React.ComponentRef<typeof SimpleBar>>(null);

  const rowVirtualizer = useVirtualizer({
    count: listState.list?.length || 0,
    getScrollElement: () => parentRef.current?.getScrollElement() || null,
    estimateSize: () => 58,
    overscan: 5,
  });
```

![CleanShot 2024-04-26 at 11 48 44](https://github.com/TanStack/virtual/assets/139225486/857d0df5-9d45-494b-855b-972b1cdfea69)


When i use normal div, it don't have the stuck.


![CleanShot 2024-04-26 at 11 47 51](https://github.com/TanStack/virtual/assets/139225486/45bf903d-0285-4248-9b42-35318e126f74)

**Reply by DmitriyWebDev** [reply]
I had the same problem today but with [overlayscrollbars-vue](https://github.com/KingSora/OverlayScrollbars/tree/master/packages/overlayscrollbars-vue) 

I just increased the **overscan** parameter to 25 and the problem was solved.
